### ISSUE TITLE
Low resolution on HD streams since v2.18.2

### ISSUE THREAD

**Issue by donchummar** [androidx/media#693]
### Version

Media3 1.1.1

### More version details

I observed the identical behavior described in this GitHub issue: https://github.com/google/ExoPlayer/issues/10898. It appears that you maintain a list of devices that bypass certain performance points. I conducted a test by including the device model in the list, and this resolved the issue. I am including the relevant code below for your reference.

MediaCodecInfo

 ``` /** Whether the device is known to have wrong {@link PerformancePoint} declarations. */
  private static boolean needsIgnorePerformancePointsWorkaround() {
    // See https://github.com/google/ExoPlayer/issues/10898 and [internal ref: b/267324685].
    return /* Chromecast with Google TV */ Util.DEVICE.equals("sabrina")
        || Util.DEVICE.equals("boreal")
        /* Lenovo Tablet M10 FHD Plus */
        || Util.MODEL.startsWith("Lenovo TB-X605")
        || Util.MODEL.startsWith("Lenovo TB-X606")
        || Util.MODEL.startsWith("Lenovo TB-X616")
     // Added by me 
        || Util.MODEL.startsWith("SM-T733");
  }  
```

### Devices that reproduce the issue

Galaxy Tab S7 FE 
Model name : SM-T733

### Reproducible in the demo app?

Not tested

### Reproduction steps

Play widewine drm asset with more than 30fps 

### Expected result

It should select max resolution available 

### Actual result

Stuck with lowest resolution available 

### Media

.

### Bug Report

- [ ] You will email the zip file produced by `adb bugreport` to android-media-github@google.com after filing this issue.

**Comment by microkatz** [comment]
Hi @donchummar,

Could you provide additional details of the content you were trying to play? Can you provide the resolutions and framerates that were available. The resolution and framerate you expected to be playable and the actual resultant selection? 

Thanks!

**Comment by donchummar** [comment]
The content i am playing having multiple resolution available 
960x540 with framerate 50/1 
960x540 with framerate 50/1 
960x540 with framerate 50/1 
960x540 with framerate 50/1 
960x540 with framerate 50/1

**Comment by tidoemanuele** [comment]
@donchummar Unfortunately, if you add the Galaxy S7 to that function, it may attempt to play content at the highest resolution. This could lead to several issues, including overheating, excessive battery drain, and, of course, rendering performance problems. I hope that the `needsIgnorePerformancePointsWorkaround` will be eliminated and replaced with a more refined logic. Ideally, this new logic would allow us to set a tolerance value that would give some consideration to the PerformancePoints without completely disregarding them. This is significant because the Galaxy S7 or another device may handle 1280x720@50 smoothly, but could encounter difficulties when attempting to handle 1920x1080@50.

**Comment by donchummar** [comment]
@tidoemanuele  Can you provide a approximate time frame of getting it fixed with a more refined logic. because we are blocked by this issue for going with our next release since galaxy s7 is one of the popular device on market. 

Thanks

**Comment by tidoemanuele** [comment]
@donchummar I wrote my own CustomMediaCodecVideoRenderer following the @tonihei explanation on how I can do it my way:
[](https://github.com/google/ExoPlayer/issues/10898#issuecomment-1721390605)
But this is not the player team vision so as you I would like to have an official solution coming directly from the player team.

**Comment by microkatz** [comment]
@donchummar 

Thank you for your patience and continued discussion here for a solution. We are still working on figuring out an official solution for this type of problem.

One thing that could help is what was the encoding of the content you were trying to play? You described the resolutions and framerates [here](https://github.com/androidx/media/issues/693#issuecomment-1749834010) but did not say what codec was of your content.

**Comment by donchummar** [comment]
we have been trying to play dash avc and stuck on lowest resoluton

**Comment by microkatz** [comment]
Hello @donchummar,

I pushed the commit https://github.com/androidx/media/commit/4515a0c3f24706a43b3247b558b14d98f2b0fce2 into the main branch that should hopefully address your issue. Please let me know if that helps!

**Comment by donchummar** [comment]
@microkatz  Thanks you so much for the fix , i have verified and it working perfectly fine with this commit, does it comes with upcoming 1.2.0 release ?

**Comment by microkatz** [comment]
@donchummar 

I am sorry but the 1.2.0 release had already been cut. The commit may be pulled into a bug fix release of 1.2.1 and I can message here if that becomes the the case.

**Comment by microkatz** [comment]
I'm going to close this issue as the fix did solve the problem. Please open up a new issue if it did not or you have additional questions.
